frontend/src/concepts/pipelines/content/K8sStatusIcon.tsx

```diff
@@ -0,0 +1,62 @@
+import * as React from 'react';
+import {
+  PendingIcon,
+  InProgressIcon,
+  CheckCircleIcon,
+  ExclamationTriangleIcon,
+  ExclamationCircleIcon,
+} from '@patternfly/react-icons';
+import { Icon } from '@patternfly/react-core';
+
+export enum StatusType {
+  PENDING = 'pending',
+  IN_PROGRESS = 'in-progress',
+  SUCCESS = 'success',
+  WARNING = 'warning',
+  ERROR = 'error',
+}
+
+type StatusConfig = {
+  icon: React.ComponentType<{ style?: React.CSSProperties; className?: string }>;
+  statusColor?: 'success' | 'danger' | 'warning' | 'info';
+  className?: string;
+};
+
+const STATUS_CONFIG: Record<StatusType, StatusConfig> = {
+  [StatusType.PENDING]: {
+    icon: PendingIcon,
+  },
+  [StatusType.IN_PROGRESS]: {
+    icon: InProgressIcon,
+    className: 'odh-u-spin',
+  },
+  [StatusType.SUCCESS]: {
+    icon: CheckCircleIcon,
+    statusColor: 'success',
+  },
+  [StatusType.WARNING]: {
+    icon: ExclamationTriangleIcon,
+    statusColor: 'warning',
+  },
+  [StatusType.ERROR]: {
+    icon: ExclamationCircleIcon,
+    statusColor: 'danger',
+  },
+};
+
+type StatusIconProps = {
+  status: StatusType;
+};
+
+const K8sStatusIcon: React.FC<StatusIconProps> = ({ status }) => {
```

Comment:
not really a k8s status. it is more specifically a pipeline server condition status. we should rename so people dont end up using this outside of that specific CRD's conditions